app/components/metadata-panel.tsx

```diff
@@ -514,37 +503,70 @@ const TabPanelDataDimension = ({
       description: dim.description,
       showExpandButton: false,
     };
-  }, [dim.description, expandable, dim.iri, selectedDimension]);
+  }, [dim.description, expanded]);
   const iconName = React.useMemo(() => {
     return getDimensionIconName(dim.__typename);
   }, [dim.__typename]);
 
   const handleClick = React.useCallback(() => {
-    if (expandable) {
+    if (!expanded) {
       setSelectedDimension(dim);
     }
-  }, [expandable, dim, setSelectedDimension]);
+  }, [expanded, dim, setSelectedDimension]);
 
   return (
     <div>
-      <Flex>
-        <Button
-          className={classes.dimensionButton}
-          variant="text"
-          size="small"
-          onClick={handleClick}
-          sx={{
-            ":hover": { color: expandable ? "primary.main" : "grey.800" },
-            cursor: expandable ? "pointer" : "default",
-          }}
-        >
-          <Icon className={classes.dimensionIcon} name={iconName} />
-          <Typography variant="body2" fontWeight="bold">
-            {dim.label}
-          </Typography>
-        </Button>
+      <Flex sx={{ justifyContent: "space-between" }}>
+        <div>
+          <Button
+            className={classes.dimensionButton}
+            variant="text"
+            size="small"
+            onClick={handleClick}
+            sx={{
+              ":hover": { color: !expanded ? "primary.main" : "grey.800" },
+              cursor: !expanded ? "pointer" : "default",
+            }}
+          >
+            <Typography variant="body2" fontWeight="bold">
+              {dim.label}
+            </Typography>
+          </Button>
+          {description && (
+            <Typography variant="body2">{description}</Typography>
+          )}
+        </div>
+        <Icon name={iconName} />
       </Flex>
-      {description && <Typography variant="body2">{description}</Typography>}
+
+      <AnimatePresence>
+        {expanded && (
+          <MotionBox
+            key="dimension-values"
+            className={classes.dimensionValues}
+            component={Flex}
+            {...animationProps}
+          >
+            {dim.values.map((d) => (
+              <React.Fragment key={d.value}>
+                <Typography variant="body2" {...animationProps}>
+                  {d.label}{" "}
+                  {d.alternateName ? (
+                    <span style={{ fontStyle: "italic" }}>
+                      ({d.alternateName})
+                    </span>
+                  ) : (
+                    ""
+                  )}
+                </Typography>
+                {d.description ? (
+                  <Typography variant="caption">{d.description}</Typography>
+                ) : null}
+              </React.Fragment>
+            ))}
+          </MotionBox>
+        )}
+      </AnimatePresence>
```

Comment:
I feel like it'd be interesting to have the dimension values or for example the min/max as a gray text under the dimension name in the panel describing all the dimensions (so that you do not have to open the panel in most cases, and also to give a taste to the user of what she will find in the details). This is not for the PR, but something to keep in mind. @AnninaWalker